java_api/standard_api/src/main/java/uk/ramp/api/StandardApi.java

```diff
@@ -1,5 +1,127 @@
+import uk.ramp.parameters.ParameterDataWriterImpl;
+import uk.ramp.samples.Samples;
+import uk.ramp.distribution.Distribution;
+import uk.ramp.file.CleanableFileChannel;
+import uk.ramp.metadata.ImmutableMetadataItem;
+import uk.ramp.parameters.ParameterDataReader;
+import uk.ramp.parameters.ParameterDataReaderImpl;
+import uk.ramp.toml.TOMLMapper;
+import uk.ramp.toml.TomlReader;
+import uk.ramp.toml.TomlWriter;
+
 public class StandardApi {
+  private final FileApi fileApi;
+  private final ParameterDataReader parameterDataReader;
+  private final ParameterDataWriter parameterDataWriter;
+
+  public StandardApi(Path configPath) {
+    this.fileApi = new FileApi(configPath);
+    this.parameterDataReader = new ParameterDataReaderImpl(new TomlReader(new TOMLMapper()));
+    this.parameterDataWriter = new ParameterDataWriterImpl(new TomlWriter(new TOMLMapper()));
+  }
+  StandardApi(FileApi fileApi, ParameterDataReader parameterDataReader, ParameterDataWriter parameterDataWriter) {
+    this.fileApi = fileApi;
+    this.parameterDataReader = parameterDataReader;
+    this.parameterDataWriter = parameterDataWriter;
+  }
+
+  public Number readEstimate(String dataProduct, String component) {
+    var query = ImmutableMetadataItem.builder()
+        .dataProduct(dataProduct)
+        .component(component)
+        .build();
+
+    ReadComponent data;
+    try(CleanableFileChannel fileChannel = fileApi.openForRead(query)) {
+      data = parameterDataReader.read(fileChannel, component);
+    }
+    return data.getEstimate();
+  }
+
+  public void writeEstimate(String dataProduct, String component, Number estimateNumber) {
+    var query = ImmutableMetadataItem.builder()
+        .dataProduct(dataProduct)
+        .component(component)
+        .extension("toml")
+        .build();
+    var estimate = ImmutableEstimate.builder().internalValue(estimateNumber).build();
+
+    try(CleanableFileChannel fileChannel = fileApi.openForWrite(query)) {
+      parameterDataWriter.write(fileChannel, component, estimate);
+    }
+  }
+
+  public Distribution readDistribution(String dataProduct, String component) {
+    var query = ImmutableMetadataItem.builder()
+        .dataProduct(dataProduct)
+        .component(component)
+        .build();
+
+    ReadComponent data;
+    try(CleanableFileChannel fileChannel = fileApi.openForRead(query)) {
+      data = parameterDataReader.read(fileChannel, component);
+    }
+    return data.getDistribution();
+  }
+
+  public void writeDistribution(String dataProduct, String component, Distribution distribution) {
+    var query = ImmutableMetadataItem.builder()
+        .dataProduct(dataProduct)
+        .component(component)
+        .extension("toml")
+        .build();
+
+    try(CleanableFileChannel fileChannel = fileApi.openForWrite(query)) {
+      parameterDataWriter.write(fileChannel, component, distribution);
+    }
+  }
+
+  public Number readSample(String dataProduct, String component) {
```

Comment:
This has changed in the spec to `readSamples` - see [this issue](https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/640).

Reply:
Thanks @mrow84. Updated PR. Please review again, thanks